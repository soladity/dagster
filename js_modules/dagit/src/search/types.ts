export enum SearchResultType {
  Asset,
  Page,
  Pipeline,
  Repository,
  Run,
  Schedule,
  Sensor,
  Solid,
}

export type SearchResult = {
  key: string;
  label: string;
  description: string;
  href: string;
  type: SearchResultType;
  tags?: string;
};
