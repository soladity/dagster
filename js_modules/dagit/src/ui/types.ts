export type Spacing = 0 | 1 | 2 | 4 | 8 | 12 | 16 | 20 | 24 | 32 | 48 | 64;
export type DirectionalSpacing =
  | Spacing
  | {
      top?: Spacing;
      right?: Spacing;
      bottom?: Spacing;
      left?: Spacing;
      vertical?: Spacing;
      horizontal?: Spacing;
    };

export type AlignItems =
  | 'stretch'
  | 'center'
  | 'start'
  | 'end'
  | 'flex-start'
  | 'flex-end'
  | 'baseline';

export type FlexDirection = 'row' | 'row-reverse' | 'column' | 'column-reverse';
export type FlexProperties = {
  alignItems?: AlignItems;
  basis?: string;
  direction?: FlexDirection;
  display?: 'flex' | 'inline-flex';
  grow?: number;
  justifyContent?:
    | 'center'
    | 'start'
    | 'end'
    | 'flex-start'
    | 'flex-end'
    | 'left'
    | 'right'
    | 'space-between'
    | 'space-around'
    | 'space-evenly'
    | 'stretch';
  shrink?: number;
  wrap?: 'nowrap' | 'wrap' | 'wrap-reverse';
};

export type BorderSide = 'top' | 'right' | 'bottom' | 'left' | 'horizontal' | 'vertical' | 'all';
export type BorderWidth = 1 | 2;
export type BorderColor = 'light' | 'medium' | 'dark';
export type BorderSetting = {width: BorderWidth; color: BorderColor; side: BorderSide};
