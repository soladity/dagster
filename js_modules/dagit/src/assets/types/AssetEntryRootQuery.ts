// @generated
/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AssetKeyInput } from "./../../types/globalTypes";

// ====================================================
// GraphQL query operation: AssetEntryRootQuery
// ====================================================

export interface AssetEntryRootQuery_assetOrError_AssetNotFoundError {
  __typename: "AssetNotFoundError";
}

export interface AssetEntryRootQuery_assetOrError_AssetsNotSupportedError {
  __typename: "AssetsNotSupportedError";
  message: string;
}

export interface AssetEntryRootQuery_assetOrError_Asset_key {
  __typename: "AssetKey";
  path: string[];
}

export interface AssetEntryRootQuery_assetOrError_Asset {
  __typename: "Asset";
  key: AssetEntryRootQuery_assetOrError_Asset_key;
}

export type AssetEntryRootQuery_assetOrError = AssetEntryRootQuery_assetOrError_AssetNotFoundError | AssetEntryRootQuery_assetOrError_AssetsNotSupportedError | AssetEntryRootQuery_assetOrError_Asset;

export interface AssetEntryRootQuery {
  assetOrError: AssetEntryRootQuery_assetOrError;
}

export interface AssetEntryRootQueryVariables {
  assetKey: AssetKeyInput;
}
