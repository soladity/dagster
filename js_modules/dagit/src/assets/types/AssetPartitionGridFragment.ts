// @generated
/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL fragment: AssetPartitionGridFragment
// ====================================================

export interface AssetPartitionGridFragment_runOrError_PipelineRunNotFoundError {
  __typename: "PipelineRunNotFoundError" | "PythonError";
}

export interface AssetPartitionGridFragment_runOrError_PipelineRun {
  __typename: "PipelineRun";
  pipelineSnapshotId: string | null;
}

export type AssetPartitionGridFragment_runOrError = AssetPartitionGridFragment_runOrError_PipelineRunNotFoundError | AssetPartitionGridFragment_runOrError_PipelineRun;

export interface AssetPartitionGridFragment {
  __typename: "AssetMaterialization";
  partition: string | null;
  runOrError: AssetPartitionGridFragment_runOrError;
}
