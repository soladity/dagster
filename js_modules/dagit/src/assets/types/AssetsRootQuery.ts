// @generated
/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: AssetsRootQuery
// ====================================================

export interface AssetsRootQuery_assetsOrError_AssetsNotSupportedError {
  __typename: "AssetsNotSupportedError";
  message: string;
}

export interface AssetsRootQuery_assetsOrError_AssetConnection_nodes_key {
  __typename: "AssetKey";
  path: string[];
}

export interface AssetsRootQuery_assetsOrError_AssetConnection_nodes {
  __typename: "Asset";
  key: AssetsRootQuery_assetsOrError_AssetConnection_nodes_key;
}

export interface AssetsRootQuery_assetsOrError_AssetConnection {
  __typename: "AssetConnection";
  nodes: AssetsRootQuery_assetsOrError_AssetConnection_nodes[];
}

export type AssetsRootQuery_assetsOrError = AssetsRootQuery_assetsOrError_AssetsNotSupportedError | AssetsRootQuery_assetsOrError_AssetConnection;

export interface AssetsRootQuery {
  assetsOrError: AssetsRootQuery_assetsOrError;
}
