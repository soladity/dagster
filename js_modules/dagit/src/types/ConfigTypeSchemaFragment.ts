/* tslint:disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL fragment: ConfigTypeSchemaFragment
// ====================================================

export interface ConfigTypeSchemaFragment_EnumConfigType_innerTypes_EnumConfigType_innerTypes {
  __typename: "EnumConfigType" | "CompositeConfigType" | "RegularConfigType" | "ListConfigType" | "NullableConfigType";
  key: string;
}

export interface ConfigTypeSchemaFragment_EnumConfigType_innerTypes_EnumConfigType {
  __typename: "EnumConfigType" | "RegularConfigType" | "ListConfigType" | "NullableConfigType";
  key: string;
  name: string | null;
  description: string | null;
  isList: boolean;
  isNullable: boolean;
  isSelector: boolean;
  innerTypes: ConfigTypeSchemaFragment_EnumConfigType_innerTypes_EnumConfigType_innerTypes[];
}

export interface ConfigTypeSchemaFragment_EnumConfigType_innerTypes_CompositeConfigType_innerTypes {
  __typename: "EnumConfigType" | "CompositeConfigType" | "RegularConfigType" | "ListConfigType" | "NullableConfigType";
  key: string;
}

export interface ConfigTypeSchemaFragment_EnumConfigType_innerTypes_CompositeConfigType_fields_configType {
  __typename: "EnumConfigType" | "CompositeConfigType" | "RegularConfigType" | "ListConfigType" | "NullableConfigType";
  key: string;
}

export interface ConfigTypeSchemaFragment_EnumConfigType_innerTypes_CompositeConfigType_fields {
  __typename: "ConfigTypeField";
  name: string;
  description: string | null;
  isOptional: boolean;
  configType: ConfigTypeSchemaFragment_EnumConfigType_innerTypes_CompositeConfigType_fields_configType;
}

export interface ConfigTypeSchemaFragment_EnumConfigType_innerTypes_CompositeConfigType {
  __typename: "CompositeConfigType";
  key: string;
  name: string | null;
  description: string | null;
  isList: boolean;
  isNullable: boolean;
  isSelector: boolean;
  innerTypes: ConfigTypeSchemaFragment_EnumConfigType_innerTypes_CompositeConfigType_innerTypes[];
  fields: ConfigTypeSchemaFragment_EnumConfigType_innerTypes_CompositeConfigType_fields[];
}

export type ConfigTypeSchemaFragment_EnumConfigType_innerTypes = ConfigTypeSchemaFragment_EnumConfigType_innerTypes_EnumConfigType | ConfigTypeSchemaFragment_EnumConfigType_innerTypes_CompositeConfigType;

export interface ConfigTypeSchemaFragment_EnumConfigType {
  __typename: "EnumConfigType" | "RegularConfigType" | "ListConfigType" | "NullableConfigType";
  key: string;
  name: string | null;
  description: string | null;
  isList: boolean;
  isNullable: boolean;
  isSelector: boolean;
  innerTypes: ConfigTypeSchemaFragment_EnumConfigType_innerTypes[];
}

export interface ConfigTypeSchemaFragment_CompositeConfigType_innerTypes_EnumConfigType_innerTypes {
  __typename: "EnumConfigType" | "CompositeConfigType" | "RegularConfigType" | "ListConfigType" | "NullableConfigType";
  key: string;
}

export interface ConfigTypeSchemaFragment_CompositeConfigType_innerTypes_EnumConfigType {
  __typename: "EnumConfigType" | "RegularConfigType" | "ListConfigType" | "NullableConfigType";
  key: string;
  name: string | null;
  description: string | null;
  isList: boolean;
  isNullable: boolean;
  isSelector: boolean;
  innerTypes: ConfigTypeSchemaFragment_CompositeConfigType_innerTypes_EnumConfigType_innerTypes[];
}

export interface ConfigTypeSchemaFragment_CompositeConfigType_innerTypes_CompositeConfigType_innerTypes {
  __typename: "EnumConfigType" | "CompositeConfigType" | "RegularConfigType" | "ListConfigType" | "NullableConfigType";
  key: string;
}

export interface ConfigTypeSchemaFragment_CompositeConfigType_innerTypes_CompositeConfigType_fields_configType {
  __typename: "EnumConfigType" | "CompositeConfigType" | "RegularConfigType" | "ListConfigType" | "NullableConfigType";
  key: string;
}

export interface ConfigTypeSchemaFragment_CompositeConfigType_innerTypes_CompositeConfigType_fields {
  __typename: "ConfigTypeField";
  name: string;
  description: string | null;
  isOptional: boolean;
  configType: ConfigTypeSchemaFragment_CompositeConfigType_innerTypes_CompositeConfigType_fields_configType;
}

export interface ConfigTypeSchemaFragment_CompositeConfigType_innerTypes_CompositeConfigType {
  __typename: "CompositeConfigType";
  key: string;
  name: string | null;
  description: string | null;
  isList: boolean;
  isNullable: boolean;
  isSelector: boolean;
  innerTypes: ConfigTypeSchemaFragment_CompositeConfigType_innerTypes_CompositeConfigType_innerTypes[];
  fields: ConfigTypeSchemaFragment_CompositeConfigType_innerTypes_CompositeConfigType_fields[];
}

export type ConfigTypeSchemaFragment_CompositeConfigType_innerTypes = ConfigTypeSchemaFragment_CompositeConfigType_innerTypes_EnumConfigType | ConfigTypeSchemaFragment_CompositeConfigType_innerTypes_CompositeConfigType;

export interface ConfigTypeSchemaFragment_CompositeConfigType_fields_configType {
  __typename: "EnumConfigType" | "CompositeConfigType" | "RegularConfigType" | "ListConfigType" | "NullableConfigType";
  key: string;
}

export interface ConfigTypeSchemaFragment_CompositeConfigType_fields {
  __typename: "ConfigTypeField";
  name: string;
  description: string | null;
  isOptional: boolean;
  configType: ConfigTypeSchemaFragment_CompositeConfigType_fields_configType;
}

export interface ConfigTypeSchemaFragment_CompositeConfigType {
  __typename: "CompositeConfigType";
  key: string;
  name: string | null;
  description: string | null;
  isList: boolean;
  isNullable: boolean;
  isSelector: boolean;
  innerTypes: ConfigTypeSchemaFragment_CompositeConfigType_innerTypes[];
  fields: ConfigTypeSchemaFragment_CompositeConfigType_fields[];
}

export type ConfigTypeSchemaFragment = ConfigTypeSchemaFragment_EnumConfigType | ConfigTypeSchemaFragment_CompositeConfigType;
