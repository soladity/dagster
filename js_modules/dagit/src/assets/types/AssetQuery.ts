// @generated
/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AssetKeyInput, PipelineRunStatus } from "./../../types/globalTypes";

// ====================================================
// GraphQL query operation: AssetQuery
// ====================================================

export interface AssetQuery_assetOrError_AssetNotFoundError {
  __typename: "AssetNotFoundError";
}

export interface AssetQuery_assetOrError_Asset_key {
  __typename: "AssetKey";
  path: string[];
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_runOrError_PipelineRunNotFoundError {
  __typename: "PipelineRunNotFoundError" | "PythonError";
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_runOrError_PipelineRun {
  __typename: "PipelineRun";
  id: string;
  runId: string;
  mode: string;
  status: PipelineRunStatus;
  pipelineName: string;
  pipelineSnapshotId: string | null;
}

export type AssetQuery_assetOrError_Asset_assetMaterializations_runOrError = AssetQuery_assetOrError_Asset_assetMaterializations_runOrError_PipelineRunNotFoundError | AssetQuery_assetOrError_Asset_assetMaterializations_runOrError_PipelineRun;

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_stepStats {
  __typename: "PipelineRunStepStats";
  endTime: number | null;
  startTime: number | null;
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventPathMetadataEntry {
  __typename: "EventPathMetadataEntry";
  label: string;
  description: string | null;
  path: string;
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventJsonMetadataEntry {
  __typename: "EventJsonMetadataEntry";
  label: string;
  description: string | null;
  jsonString: string;
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventUrlMetadataEntry {
  __typename: "EventUrlMetadataEntry";
  label: string;
  description: string | null;
  url: string;
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventTextMetadataEntry {
  __typename: "EventTextMetadataEntry";
  label: string;
  description: string | null;
  text: string;
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventMarkdownMetadataEntry {
  __typename: "EventMarkdownMetadataEntry";
  label: string;
  description: string | null;
  mdStr: string;
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventPythonArtifactMetadataEntry {
  __typename: "EventPythonArtifactMetadataEntry";
  label: string;
  description: string | null;
  module: string;
  name: string;
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventFloatMetadataEntry {
  __typename: "EventFloatMetadataEntry";
  label: string;
  description: string | null;
  floatValue: number;
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventIntMetadataEntry {
  __typename: "EventIntMetadataEntry";
  label: string;
  description: string | null;
  intValue: number;
}

export type AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries = AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventPathMetadataEntry | AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventJsonMetadataEntry | AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventUrlMetadataEntry | AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventTextMetadataEntry | AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventMarkdownMetadataEntry | AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventPythonArtifactMetadataEntry | AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventFloatMetadataEntry | AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventIntMetadataEntry;

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization {
  __typename: "Materialization";
  label: string;
  description: string | null;
  metadataEntries: AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries[];
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_assetLineage_assetKey {
  __typename: "AssetKey";
  path: string[];
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_assetLineage {
  __typename: "AssetLineageInfo";
  assetKey: AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_assetLineage_assetKey;
  partitions: string[];
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent {
  __typename: "StepMaterializationEvent";
  runId: string;
  timestamp: string;
  stepKey: string | null;
  stepStats: AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_stepStats;
  materialization: AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization;
  assetLineage: AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_assetLineage[];
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations {
  __typename: "AssetMaterialization";
  partition: string | null;
  runOrError: AssetQuery_assetOrError_Asset_assetMaterializations_runOrError;
  materializationEvent: AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent;
}

export interface AssetQuery_assetOrError_Asset {
  __typename: "Asset";
  key: AssetQuery_assetOrError_Asset_key;
  assetMaterializations: AssetQuery_assetOrError_Asset_assetMaterializations[];
}

export type AssetQuery_assetOrError = AssetQuery_assetOrError_AssetNotFoundError | AssetQuery_assetOrError_Asset;

export interface AssetQuery {
  assetOrError: AssetQuery_assetOrError;
}

export interface AssetQueryVariables {
  assetKey: AssetKeyInput;
}
