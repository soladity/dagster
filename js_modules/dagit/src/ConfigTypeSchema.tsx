import * as React from "react";
import gql from "graphql-tag";
import styled from "styled-components";
import { Colors } from "@blueprintjs/core";
import { ConfigTypeSchemaFragment } from "./types/ConfigTypeSchemaFragment";
import { ConfigTypeInfoFragment } from "./types/ConfigTypeInfoFragment";

type ConfigTypeSchemaTheme = "dark" | "light";

interface ConfigTypeSchemaProps {
  type: ConfigTypeSchemaFragment;
  allInnerTypes: ConfigTypeSchemaFragment[];
  theme?: ConfigTypeSchemaTheme;
  maxDepth?: number;
}

function renderTypeRecursive(
  type: ConfigTypeInfoFragment,
  typeLookup: { [typeName: string]: ConfigTypeInfoFragment },
  depth: number,
  props: ConfigTypeSchemaProps
): React.ReactElement<HTMLElement> {
  if ("fields" in type && props.maxDepth && depth === props.maxDepth) {
    return <span>...</span>;
  }
  if ("fields" in type) {
    const innerIndent = "  ".repeat(depth + 1);
    return (
      <>
        {`{`}
        {type.isSelector && (
          <DictBlockComment
            indent={innerIndent}
            content={`One of the following:`}
          />
        )}
        {type.fields.map(fieldData => (
          <DictEntry key={fieldData.name}>
            <DictBlockComment
              indent={innerIndent}
              content={fieldData.description}
            />
            {innerIndent}
            <DictKey theme={props.theme}>{fieldData.name}</DictKey>
            {fieldData.isOptional && Optional}
            {`: `}
            {renderTypeRecursive(
              typeLookup[fieldData.configType.key],
              typeLookup,
              depth + 1,
              props
            )}
          </DictEntry>
        ))}
        {"  ".repeat(depth) + "}"}
      </>
    );
  }
  if (type.isList) {
    const innerType = type.innerTypes[0].key;
    return (
      <>
        [{renderTypeRecursive(typeLookup[innerType], typeLookup, depth, props)}]
      </>
    );
  }
  if (type.isNullable) {
    const innerType = type.innerTypes[0].key;
    return (
      <>
        {renderTypeRecursive(typeLookup[innerType], typeLookup, depth, props)}
        {Optional}
      </>
    );
  }

  return <span>{type.name || "Anonymous Type"}</span>;
}

export default class ConfigTypeSchema extends React.Component<
  ConfigTypeSchemaProps
> {
  static fragments = {
    ConfigTypeSchemaFragment: gql`
      fragment ConfigTypeInfoFragment on ConfigType {
        key
        name
        description
        isList
        isNullable
        isSelector
        innerTypes {
          key
        }
        ... on CompositeConfigType {
          fields {
            name
            description
            isOptional
            configType {
              key
            }
          }
        }
      }

      fragment ConfigTypeSchemaFragment on ConfigType {
        ...ConfigTypeInfoFragment
        innerTypes {
          key
        }
      }
    `
  };

  public render() {
    const { type, allInnerTypes } = this.props;

    const allInnerTypeMap = {};
    for (const innerType of allInnerTypes) {
      allInnerTypeMap[innerType.key] = innerType;
    }

    return (
      <TypeSchemaContainer>
        <DictBlockComment content={type.description} indent="" />
        {renderTypeRecursive(type, allInnerTypeMap, 0, this.props)}
      </TypeSchemaContainer>
    );
  }
}

const TypeSchemaContainer = styled.code`
  color: ${Colors.GRAY3};
  display: block;
  white-space: pre-wrap;
  font-size: smaller;
  line-height: 18px;
`;

const DictEntry = styled.div``;

const DictKey = styled.span<{ theme: ConfigTypeSchemaTheme | undefined }>`
  color: ${({ theme }) => (theme === "dark" ? Colors.WHITE : Colors.BLACK)};
`;

const DictComment = styled.div`
  /* This allows long comments to wrap as nice indented blocks, while
     copy/pasting as a single line with space-based indentation. */
  text-indent: -1.85em;
  padding-left: 1.85em;
  white-space: initial;
`;

const DictBlockComment = ({
  indent = "",
  content
}: {
  indent: string;
  content: string | null;
}) =>
  content !== null && content !== "" ? (
    <DictComment>{`${indent.replace(
      / /g,
      "\u00A0"
    )}/* ${content} */`}</DictComment>
  ) : null;

const Optional = (
  <span style={{ fontWeight: 500, color: Colors.ORANGE2 }}>?</span>
);
