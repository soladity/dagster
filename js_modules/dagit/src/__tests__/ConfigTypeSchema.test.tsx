import * as React from "react";
import * as TestRenderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

import ConfigTypeSchema from "../ConfigTypeSchema";
import { ConfigTypeSchemaFragment } from "../types/ConfigTypeSchemaFragment";

it("renders given a basic type", () => {
  const intConfigData: ConfigTypeSchemaFragment = {
    __typename: "RegularConfigType",
    key: "Int",
    name: "Int",
    description: "An int",
    isList: false,
    isSelector: false,
    isNullable: false,
    innerTypes: []
  };
  const component = TestRenderer.create(
    <BrowserRouter>
      <ConfigTypeSchema type={intConfigData} />
    </BrowserRouter>
  );
  expect(component.toJSON()).toMatchSnapshot();
});

it("renders given a complex type", () => {
  let complexConfigData: ConfigTypeSchemaFragment = {
    __typename: "CompositeConfigType",
    key: "Dict.4",
    name: "Dict.4",
    description: "",
    isList: false,
    isSelector: false,
    isNullable: false,
    fields: [
      {
        __typename: "ConfigTypeField",
        description: "",
        name: "field_one",
        configType: {
          __typename: "RegularConfigType",
          key: "String"
        },
        isOptional: false
      },
      {
        __typename: "ConfigTypeField",
        description: "",
        name: "field_two",
        configType: {
          __typename: "RegularConfigType",
          key: "String"
        },
        isOptional: false
      },
      {
        __typename: "ConfigTypeField",
        description: "",
        name: "field_three",
        configType: {
          __typename: "RegularConfigType",
          key: "String"
        },
        isOptional: false
      },
      {
        __typename: "ConfigTypeField",
        description: "",
        name: "nested_field",
        configType: {
          __typename: "CompositeConfigType",
          key: "Dict.3"
        },
        isOptional: false
      }
    ],
    innerTypes: [
      {
        __typename: "RegularConfigType",
        key: "String",
        name: "String",
        description: "",
        isList: false,
        isSelector: false,
        isNullable: false,
        innerTypes: []
      },
      {
        __typename: "RegularConfigType",
        key: "Dict.3",
        name: "Dict.3",
        description: "",
        isList: false,
        isSelector: false,
        isNullable: false,
        innerTypes: [
          {
            __typename: "RegularConfigType",
            key: "String"
          },
          {
            __typename: "RegularConfigType",
            key: "Int"
          },
          {
            __typename: "ListConfigType",
            key: "List.Nullable.Int"
          },
          {
            __typename: "NullableConfigType",
            key: "Nullable.Int"
          }
        ]
      },
      {
        __typename: "RegularConfigType",
        key: "Int",
        name: "Int",
        description: "",
        isList: false,
        isSelector: false,
        isNullable: false,
        innerTypes: []
      },
      {
        __typename: "ListConfigType",
        key: "List.Nullable.Int",
        name: "List.Nullable.Int",
        description: "",
        isList: true,
        isSelector: false,
        isNullable: false,
        innerTypes: [
          {
            __typename: "NullableConfigType",
            key: "Nullable.Int"
          },
          {
            __typename: "RegularConfigType",
            key: "Int"
          }
        ]
      },
      {
        __typename: "NullableConfigType",
        key: "Nullable.Int",
        name: "Nullable.Int",
        description: "",
        isList: false,
        isSelector: false,
        isNullable: true,
        innerTypes: [
          {
            __typename: "RegularConfigType",
            key: "Int"
          }
        ]
      }
    ]
  };
  const component = TestRenderer.create(
    <BrowserRouter>
      <ConfigTypeSchema type={complexConfigData} />
    </BrowserRouter>
  );
  expect(component.toJSON()).toMatchSnapshot();
});
