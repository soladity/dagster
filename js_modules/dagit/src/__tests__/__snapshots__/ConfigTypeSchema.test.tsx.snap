// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders given a basic type 1`] = `
<code
  className="sc-bdVaJa eOofRl"
>
  <div
    className="sc-bxivhb dlbVAt"
  >
    /* An int */
  </div>
  <span>
    Int
  </span>
</code>
`;

exports[`renders given a complex type 1`] = `
<code
  className="sc-bdVaJa eOofRl"
>
  {
  <div
    className="sc-bwzfXH uTxCW"
  >
      
    <span
      className="sc-htpNat dJsUWn"
    >
      field_one
    </span>
    : 
    <span>
      String
    </span>
  </div>
  <div
    className="sc-bwzfXH uTxCW"
  >
      
    <span
      className="sc-htpNat dJsUWn"
    >
      field_two
    </span>
    : 
    <span>
      String
    </span>
  </div>
  <div
    className="sc-bwzfXH uTxCW"
  >
      
    <span
      className="sc-htpNat dJsUWn"
    >
      field_three
    </span>
    : 
    <span>
      String
    </span>
  </div>
  <div
    className="sc-bwzfXH uTxCW"
  >
      
    <span
      className="sc-htpNat dJsUWn"
    >
      nested_field
    </span>
    : 
    <span>
      Dict.3
    </span>
  </div>
  }
</code>
`;
